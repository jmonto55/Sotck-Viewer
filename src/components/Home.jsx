import { Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { selectMarket } from '../redux/markets/marketsSlice';
import stock from '../assets/stock.svg';
import euronext from '../assets/euronext.svg';
import amex from '../assets/amex.png';
import nyse from '../assets/nyse.svg';
import nasdaq from '../assets/nasdaq.svg';
import tsx from '../assets/tsx.svg';
import etf from '../assets/etf.svg';

const Home = () => {
  const dispatch = useDispatch();

  return (
    <div className="z-10 w-screen text-white mt-24 px-5 mb-10 flex flex-col justify-center">
      <header className="m-auto max-w-screen-2xl bg-white w-full bg-opacity-10 h-48 p-2 md:p-10 rounded-lg flex justify-between items-center">
        <h2 className="animate-pulse-slow text-center md:text-2xl text-white">Welcome to Quantum Stocks</h2>
        <img src={stock} alt="stock" className="animate-pulse-slow mt-10 w-40 md:w-64" />
      </header>
      <section className="m-auto w-full max-w-screen-2xl">
        <h2 className="mt-5 p-5 bg-black rounded-t-lg opacity-70">Stocks by Market</h2>
        <div className="grid grid-cols-2">
          <Link
            to="/market"
            className="relative p-10 bg-neutral-900 h-40 flex flex-col justify-center items-center hover:opacity-60"
            type="button"
            onClick={() => dispatch(selectMarket('ETF'))}
          >
            <span className="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110">
              arrow_forward_ios
            </span>
            <img src={etf} alt="euronext" className="m-auto opacity-20 w-64 h-28 animate-pulse-slow" />
            {' '}
            ETF
          </Link>
          <Link
            to="/market"
            className="relative p-10 bg-neutral-800 h-40 flex flex-col justify-center items-center hover:opacity-60"
            type="button"
            onClick={() => dispatch(selectMarket('NYSE'))}
          >
            <span className="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110">
              arrow_forward_ios
            </span>
            <img src={nyse} alt="euronext" className="m-auto opacity-20 w-64 h-28 animate-pulse-slow" />
            {' '}
            NYSE
          </Link>
          <Link
            to="/market"
            className="relative p-10 bg-neutral-800 h-40 flex flex-col justify-center items-center hover:opacity-60"
            type="button"
            onClick={() => dispatch(selectMarket('EURONEXT'))}
          >
            <span className="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110">
              arrow_forward_ios
            </span>
            <img src={euronext} alt="euronext" className="m-auto opacity-20 w-64 h-28 animate-pulse-slow" />
            {' '}
            EURONEXT
          </Link>
          <Link
            to="/market"
            className="relative p-10 bg-neutral-900 h-40 flex flex-col justify-center items-center hover:opacity-60"
            type="button"
            onClick={() => dispatch(selectMarket('AMEX'))}
          >
            <span className="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110">
              arrow_forward_ios
            </span>
            <img src={amex} alt="euronext" className="m-auto opacity-60 h-28 animate-pulse-slow" />
            {' '}
            AMEX
          </Link>
          <Link
            to="/market"
            className="relative p-10 bg-neutral-900 h-40 flex flex-col justify-center items-center hover:opacity-60"
            type="button"
            onClick={() => dispatch(selectMarket('TSX'))}
          >
            <span className="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110">
              arrow_forward_ios
            </span>
            <img src={tsx} alt="euronext" className="m-auto opacity-20 h-20 md:h-28 w-64 animate-pulse-slow" />
            {' '}
            TSX
          </Link>
          <Link
            to="/market"
            className="relative p-10 bg-neutral-800 h-40 flex flex-col justify-center items-center hover:opacity-60"
            type="button"
            onClick={() => dispatch(selectMarket('NASDAQ'))}
          >
            <span className="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110">
              arrow_forward_ios
            </span>
            <img src={nasdaq} alt="euronext" className="m-auto opacity-20 w-64 h-28 animate-pulse-slow" />
            {' '}
            NASDAQ
          </Link>
        </div>
      </section>
    </div>
  );
};
export default Home;
