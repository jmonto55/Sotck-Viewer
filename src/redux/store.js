import { configureStore } from '@reduxjs/toolkit';
import marketsReducer from './markets/marketsSlice';

const store = configureStore({
  reducer: {
    markets: marketsReducer,
  },
});

export default store;
