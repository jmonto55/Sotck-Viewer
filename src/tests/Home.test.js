import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import Home from '../components/Home';
import store from '../redux/store';
import '@testing-library/jest-dom';

describe('Test suite for Home component', () => {
  it('Should render a Snapshot of Home component', () => {
    const footer = render(
      <Provider store={store}>
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </Provider>,
    );
    expect(footer).toMatchSnapshot();
  });
  it('Should test if the welcome message is being rendered', () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </Provider>,
    );

    expect(screen.getByText('Stocks by Market')).toBeInTheDocument();
  });

  it('Should test if the market links are being rendered', () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </Provider>,
    );

    const etfLink = screen.getByText(/etf/i);
    expect(etfLink).toBeInTheDocument();

    const nyseLink = screen.getByText(/nyse/i);
    expect(nyseLink).toBeInTheDocument();

    const euronextLink = screen.getByText(/euronext/i);
    expect(euronextLink).toBeInTheDocument();

    const amexLink = screen.getByText(/amex/i);
    expect(amexLink).toBeInTheDocument();

    const tsxLink = screen.getByText(/tsx/i);
    expect(tsxLink).toBeInTheDocument();

    const nasdaqLink = screen.getByText(/nasdaq/i);
    expect(nasdaqLink).toBeInTheDocument();
  });

  it('Should dispatch the selectMarket action when a link is clicked', () => {
    const selectMarketMock = jest.fn();
    store.dispatch = selectMarketMock;

    render(
      <Provider store={store}>
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </Provider>,
    );

    const etfLink = screen.getByText(/etf/i);
    fireEvent.click(etfLink);
    expect(selectMarketMock).toHaveBeenCalledWith({ payload: 'ETF', type: 'markets/selectMarket' });

    const nyseLink = screen.getByText(/nyse/i);
    fireEvent.click(nyseLink);
    expect(selectMarketMock).toHaveBeenCalledWith({ payload: 'NYSE', type: 'markets/selectMarket' });

    const euronextLink = screen.getByText(/euronext/i);
    fireEvent.click(euronextLink);
    expect(selectMarketMock).toHaveBeenCalledWith({ payload: 'EURONEXT', type: 'markets/selectMarket' });

    const amexLink = screen.getByText(/amex/i);
    fireEvent.click(amexLink);
    expect(selectMarketMock).toHaveBeenCalledWith({ payload: 'AMEX', type: 'markets/selectMarket' });

    const tsxLink = screen.getByText(/tsx/i);
    fireEvent.click(tsxLink);
    expect(selectMarketMock).toHaveBeenCalledWith({ payload: 'TSX', type: 'markets/selectMarket' });

    const nasdaqLink = screen.getByText(/nasdaq/i);
    fireEvent.click(nasdaqLink);
    expect(selectMarketMock).toHaveBeenCalledWith({ payload: 'NASDAQ', type: 'markets/selectMarket' });
  });
});
