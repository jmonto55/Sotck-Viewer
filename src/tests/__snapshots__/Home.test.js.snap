// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Home component Should render a Snapshot of Home component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-10 w-screen text-white mt-24 px-5 mb-10 flex flex-col justify-center"
      >
        <header
          class="m-auto max-w-screen-2xl bg-white w-full bg-opacity-10 h-48 p-2 md:p-10 rounded-lg flex justify-between items-center"
        >
          <h2
            class="animate-pulse-slow text-center md:text-2xl text-white"
          >
            Welcome to Quantum Stocks
          </h2>
          <img
            alt="stock"
            class="animate-pulse-slow mt-10 w-40 md:w-64"
            src="stock.svg"
          />
        </header>
        <section
          class="m-auto w-full max-w-screen-2xl"
        >
          <h2
            class="mt-5 p-5 bg-black rounded-t-lg opacity-70"
          >
            Stocks by Market
          </h2>
          <div
            class="grid grid-cols-2"
          >
            <a
              class="relative p-10 bg-neutral-900 h-40 flex flex-col justify-center items-center hover:opacity-60"
              href="/market"
              type="button"
            >
              <span
                class="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110"
              >
                arrow_forward_ios
              </span>
              <img
                alt="euronext"
                class="m-auto opacity-20 w-64 h-28 animate-pulse-slow"
                src="etf.svg"
              />
               
              ETF
            </a>
            <a
              class="relative p-10 bg-neutral-800 h-40 flex flex-col justify-center items-center hover:opacity-60"
              href="/market"
              type="button"
            >
              <span
                class="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110"
              >
                arrow_forward_ios
              </span>
              <img
                alt="euronext"
                class="m-auto opacity-20 w-64 h-28 animate-pulse-slow"
                src="nyse.svg"
              />
               
              NYSE
            </a>
            <a
              class="relative p-10 bg-neutral-800 h-40 flex flex-col justify-center items-center hover:opacity-60"
              href="/market"
              type="button"
            >
              <span
                class="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110"
              >
                arrow_forward_ios
              </span>
              <img
                alt="euronext"
                class="m-auto opacity-20 w-64 h-28 animate-pulse-slow"
                src="euronext.svg"
              />
               
              EURONEXT
            </a>
            <a
              class="relative p-10 bg-neutral-900 h-40 flex flex-col justify-center items-center hover:opacity-60"
              href="/market"
              type="button"
            >
              <span
                class="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110"
              >
                arrow_forward_ios
              </span>
              <img
                alt="euronext"
                class="m-auto opacity-60 h-28 animate-pulse-slow"
                src="amex.png"
              />
               
              AMEX
            </a>
            <a
              class="relative p-10 bg-neutral-900 h-40 flex flex-col justify-center items-center hover:opacity-60"
              href="/market"
              type="button"
            >
              <span
                class="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110"
              >
                arrow_forward_ios
              </span>
              <img
                alt="euronext"
                class="m-auto opacity-20 h-20 md:h-28 w-64 animate-pulse-slow"
                src="tsx.svg"
              />
               
              TSX
            </a>
            <a
              class="relative p-10 bg-neutral-800 h-40 flex flex-col justify-center items-center hover:opacity-60"
              href="/market"
              type="button"
            >
              <span
                class="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110"
              >
                arrow_forward_ios
              </span>
              <img
                alt="euronext"
                class="m-auto opacity-20 w-64 h-28 animate-pulse-slow"
                src="nasdaq.svg"
              />
               
              NASDAQ
            </a>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-10 w-screen text-white mt-24 px-5 mb-10 flex flex-col justify-center"
    >
      <header
        class="m-auto max-w-screen-2xl bg-white w-full bg-opacity-10 h-48 p-2 md:p-10 rounded-lg flex justify-between items-center"
      >
        <h2
          class="animate-pulse-slow text-center md:text-2xl text-white"
        >
          Welcome to Quantum Stocks
        </h2>
        <img
          alt="stock"
          class="animate-pulse-slow mt-10 w-40 md:w-64"
          src="stock.svg"
        />
      </header>
      <section
        class="m-auto w-full max-w-screen-2xl"
      >
        <h2
          class="mt-5 p-5 bg-black rounded-t-lg opacity-70"
        >
          Stocks by Market
        </h2>
        <div
          class="grid grid-cols-2"
        >
          <a
            class="relative p-10 bg-neutral-900 h-40 flex flex-col justify-center items-center hover:opacity-60"
            href="/market"
            type="button"
          >
            <span
              class="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110"
            >
              arrow_forward_ios
            </span>
            <img
              alt="euronext"
              class="m-auto opacity-20 w-64 h-28 animate-pulse-slow"
              src="etf.svg"
            />
             
            ETF
          </a>
          <a
            class="relative p-10 bg-neutral-800 h-40 flex flex-col justify-center items-center hover:opacity-60"
            href="/market"
            type="button"
          >
            <span
              class="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110"
            >
              arrow_forward_ios
            </span>
            <img
              alt="euronext"
              class="m-auto opacity-20 w-64 h-28 animate-pulse-slow"
              src="nyse.svg"
            />
             
            NYSE
          </a>
          <a
            class="relative p-10 bg-neutral-800 h-40 flex flex-col justify-center items-center hover:opacity-60"
            href="/market"
            type="button"
          >
            <span
              class="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110"
            >
              arrow_forward_ios
            </span>
            <img
              alt="euronext"
              class="m-auto opacity-20 w-64 h-28 animate-pulse-slow"
              src="euronext.svg"
            />
             
            EURONEXT
          </a>
          <a
            class="relative p-10 bg-neutral-900 h-40 flex flex-col justify-center items-center hover:opacity-60"
            href="/market"
            type="button"
          >
            <span
              class="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110"
            >
              arrow_forward_ios
            </span>
            <img
              alt="euronext"
              class="m-auto opacity-60 h-28 animate-pulse-slow"
              src="amex.png"
            />
             
            AMEX
          </a>
          <a
            class="relative p-10 bg-neutral-900 h-40 flex flex-col justify-center items-center hover:opacity-60"
            href="/market"
            type="button"
          >
            <span
              class="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110"
            >
              arrow_forward_ios
            </span>
            <img
              alt="euronext"
              class="m-auto opacity-20 h-20 md:h-28 w-64 animate-pulse-slow"
              src="tsx.svg"
            />
             
            TSX
          </a>
          <a
            class="relative p-10 bg-neutral-800 h-40 flex flex-col justify-center items-center hover:opacity-60"
            href="/market"
            type="button"
          >
            <span
              class="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110"
            >
              arrow_forward_ios
            </span>
            <img
              alt="euronext"
              class="m-auto opacity-20 w-64 h-28 animate-pulse-slow"
              src="nasdaq.svg"
            />
             
            NASDAQ
          </a>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
